test/Emission.test.ts

```diff
@@ -0,0 +1,129 @@
+import { expect } from 'chai';
+import hre, { ethers } from 'hardhat';
+import * as mento from '@mento-protocol/mento-sdk';
+import * as helpers from '@nomicfoundation/hardhat-toolbox/network-helpers';
+
+import {
+  Emission,
+  Emission__factory,
+  MentoToken,
+  MentoToken__factory,
+} from '@mento-protocol/mento-core-ts';
+
+describe('Emission Contract', function () {
+  const { provider } = ethers;
+
+  const DAY = 60 * 60 * 24;
+  const MONTH = 30 * DAY;
+  const YEAR = 365 * DAY;
+
+  const TOTAL_EMISSION_SUPPLY = BigInt('650000000000000000000000000');
+  const EMISSION_SCHEDULE: [string, number, BigInt][] = [
+    ['1 month', MONTH, BigInt('3692586569806444700000000')],
+    ['6 months', 6 * MONTH, BigInt('21843234320275928950000000')],
+    ['1 year', YEAR, BigInt('43528555608969174900000000')],
+    ['10 years', 10 * YEAR, BigInt('325091005879265576600000000')],
+    ['15 years', 15 * YEAR, BigInt('421181077924230273650000000')],
+    ['25 years', 25 * YEAR, BigInt('554584121910167354200000000')],
+    ['30 years', 30 * YEAR, BigInt('624618096957158932550000000')],
+    ['40 years', 40 * YEAR, BigInt(TOTAL_EMISSION_SUPPLY)],
+  ];
+
+  let governanceAddresses: mento.ContractAddresses;
+  let emission: Emission;
+  let mentoToken: MentoToken;
+
+  beforeEach(async function () {
+    // reset the fork state between tests to not pollute the state
+    // @ts-expect-error - forking doesn't exist in hre for some reason
+    await helpers.reset(hre.network.config.forking.url);
+  });
+
+  before(async function () {
+    const chainId = hre.network.config.chainId;
+    if (!chainId) {
+      throw new Error('Chain ID not found');
+    }
+
+    governanceAddresses = mento.getContractsByChainId(chainId);
```

Comment:
this is not gonna work in the new version of sdk, lets use mento.addresses[chainId]!;